stunnel/stunnel.ml

```diff
@@ -155,12 +155,15 @@ let config_file verify_cert extended_diagnosis host port legacy =
       Some x -> Printf.sprintf "TIMEOUTidle = %d" x)
     ; Printf.sprintf "connect=%s:%d" host port
     ]
-  ; if is_fips then [] else ["fips=no"]
-  ; if extended_diagnosis then ["debug=4"] else []
+  ; if is_fips then ["fips=yes"] else ["fips=no"]
+  ; if extended_diagnosis then ["debug=authpriv.7"] else ["debug=authpriv.5"]
   ; if verify_cert then
       ["verify=2"
+      ; sprintf "checkHost=%s" host
       ; sprintf "CAfile=%s" certificates_bundle_path
-      ; sprintf "CRLpath=%s" crl_path
+      ; (match Sys.readdir crl_path with
```

Comment:
What is this code supposed to do? This: 

* fails when the directory does not exist
* returns `""` when the directory is empty
* returns  a string when the directory is not empty

Would it be better to use `Sys.is_directory`?

Reply:
It looks like the 3 cases are needed because of the change of behaviour in the latest versions of stunnel, as explained in https://github.com/xapi-project/xen-api-libs-transitional/pull/71/commits/d51a66dd70efa66083e9d81c06bbe3aa1113a521

Reply:
Indeed. I pushed a fixup. I need to check:
1. if it exits
2. if it is a directory
3. if it is  empty